From 2a412d83fc519ab0c1ef5dc5aab4431a46315706 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Tue, 14 May 2024 06:55:43 +0200
Subject: [PATCH] use POSIX sh and remove some bashism

---
 src/make.bash | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/make.bash b/src/make.bash
index 76ad516..73f5a26 100755
--- a/src/make.bash
+++ b/src/make.bash
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/sh
 # Copyright 2009 The Go Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style
 # license that can be found in the LICENSE file.
@@ -167,7 +167,7 @@ nogoenv() {
 }
 
 export GOROOT="$(cd .. && pwd)"
-IFS=$'\n'; for go_exe in $(type -ap go); do
+for go_exe in $(command -v go); do
 	if [[ ! -x "$GOROOT_BOOTSTRAP/bin/go" ]]; then
 		goroot=$(GOROOT= nogoenv "$go_exe" env GOROOT)
 		if [[ "$goroot" != "$GOROOT" ]]; then
@@ -198,7 +198,9 @@ if [[ "$GOROOT_BOOTSTRAP" == "$GOROOT" ]]; then
 	exit 1
 fi
 rm -f cmd/dist/dist
+(
 GOROOT="$GOROOT_BOOTSTRAP" nogoenv "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+)
 
 # -e doesn't propagate out of eval, so check success by hand.
 eval $(./cmd/dist/dist env -p || echo FAIL=true)
-- 
2.44.0

