From 94adfcfb2933d0b053b971cba3d93dfee749b767 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Sat, 4 May 2024 20:47:09 +0200
Subject: [PATCH 3/3] add proper GNU stack markings so we dont get the default:
 executable

---
 crc_i386.S | 3 +++
 match.S    | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/crc_i386.S b/crc_i386.S
index 38dbc86..62e5c16 100644
--- a/crc_i386.S
+++ b/crc_i386.S
@@ -302,3 +302,6 @@ _crc32:                         /* ulg crc32(ulg crc, uch *buf, extent len) */
 #endif /* i386 || _i386 || _I386 || __i386 */
 
 #endif /* !USE_ZLIB && !CRC_TABLE_ONLY */
+
+.section .note.GNU-stack, "", @progbits
+.previous
diff --git a/match.S b/match.S
index eb8f735..b900d83 100644
--- a/match.S
+++ b/match.S
@@ -405,3 +405,6 @@ L__return:
 #endif /* i386 || _I386 || _i386 || __i386  */
 
 #endif /* !USE_ZLIB */
+
+.section .note.GNU-stack, "", @progbits
+.previous
-- 
2.44.0

