From eb4798bb606fcb2a76eecf8b5008d19a1053df48 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Thu, 12 Mar 2026 12:33:06 +0100
Subject: [PATCH] dont set path to config.h so we can override it

---
 src/hevel.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hevel.h b/src/hevel.h
index 466aee7..491d34f 100644
--- a/src/hevel.h
+++ b/src/hevel.h
@@ -23,7 +23,7 @@
 #include <swc.h>
 #include <xkbcommon/xkbcommon-keysyms.h>
 
-#include "../config.h"
+#include "config.h"
 #include "nein_cursor.h"
 
 #include "input.h"
-- 
2.49.0

