From cd4785280b4c02d1c71ea30d9278fea8f8bdca65 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Sun, 3 Dec 2023 16:36:54 +0100
Subject: [PATCH] add bookmarking using bmks

---
 config.def.h | 18 ++++++++++++++++++
 surf.1       |  9 +++++++++
 2 files changed, 27 insertions(+)

diff --git a/config.def.h b/config.def.h
index 8057e88..99dbf50 100644
--- a/config.def.h
+++ b/config.def.h
@@ -81,6 +81,20 @@ static WebKitFindOptions findopts = WEBKIT_FIND_OPTIONS_CASE_INSENSITIVE |
         } \
 }
 
+/* bookmarks, need bmks script */
+#define BM_PICK(p, q) { .v = (char *[]){ "/bin/sh", "-c", \
+        "prop=\"$(bmks)\" &&" \
+        "xprop -id $2 -f $1 8s -set $1 \"$prop\"", \
+        p, q, winid, NULL } }
+
+#define BM_ADD { .v = (char *[]){ "/bin/sh", "-c", \
+        "bmks a \"$(xprop -id $0 _SURF_URI | cut -d '\"' -f 2)\"", \
+        winid, NULL } }
+
+#define BM_DEL { .v = (char *[]){ "/bin/sh", "-c", \
+        "bmks d", \
+        winid, NULL } }
+
 /* DOWNLOAD(URI, referer) */
 #define DOWNLOAD(u, r) { \
         .v = (const char *[]){ "st", "-e", "/bin/sh", "-c",\
@@ -139,6 +153,10 @@ static Key keys[] = {
 	{ MODKEY,                GDK_KEY_f,      spawn,      SETPROP("_SURF_FIND", "_SURF_FIND", PROMPT_FIND) },
 	{ MODKEY,                GDK_KEY_slash,  spawn,      SETPROP("_SURF_FIND", "_SURF_FIND", PROMPT_FIND) },
 
+	{ MODKEY,                GDK_KEY_o,      spawn,      BM_PICK("_SURF_URI", "_SURF_GO") },
+	{ MODKEY,                GDK_KEY_a,      spawn,      BM_ADD },
+	{ MODKEY,				 GDK_KEY_d,      spawn,      BM_DEL },
+
 	{ 0,                     GDK_KEY_Escape, stop,       { 0 } },
 	{ MODKEY,                GDK_KEY_c,      stop,       { 0 } },
 	
diff --git a/surf.1 b/surf.1
index 78e8f9c..d9734cc 100644
--- a/surf.1
+++ b/surf.1
@@ -158,6 +158,15 @@ Scroll horizontally to the right.
 .B Ctrl\-u
 Scroll horizontally to the left.
 .TP
+.B Ctrl\-o
+Show bookmarks.
+.TP
+.B Ctrl\-a
+Bookmark current url.
+.TP
+.B Ctrl\-d
+Delete bookmark.
+.TP
 .B Ctrl\-Shift\-k or Ctrl\-+
 Zooms page in.
 .TP
-- 
2.43.0

