From 52e195daf74388d29d3b20d1ffd9ca3ffbfe929f Mon Sep 17 00:00:00 2001
From: dalem <lain@dalem.foo>
Date: Tue, 27 Jan 2026 17:36:27 +0100
Subject: [PATCH] add fullscreen support

---
 libswc/window.c | 21 ++++++++++++++++-----
 libswc/window.h |  5 +++++
 2 files changed, 21 insertions(+), 5 deletions(-)

diff --git a/libswc/window.c b/libswc/window.c
index f46793d..84e97b0 100644
--- a/libswc/window.c
+++ b/libswc/window.c
@@ -194,12 +194,23 @@ EXPORT void
 swc_window_set_fullscreen(struct swc_window *base, struct swc_screen *screen)
 {
 	struct window *window = INTERNAL(base);
+	struct swc_rectangle geom;
+	swc_window_get_geometry(base, &geom);
+	
+	if (window->mode != WINDOW_MODE_FULLSCREEN) {
+		window->prev.geom = geom;
+		window->prev.mode = window->mode;
+		swc_window_set_geometry(base, &screen->usable_geometry);
+
+		if (window->impl->set_mode)
+			window->impl->set_mode(window, WINDOW_MODE_FULLSCREEN);
+		window->mode = WINDOW_MODE_FULLSCREEN;
+	}
 
-	/* TODO: Implement fullscreen windows. */
-
-	if (window->impl->set_mode)
-		window->impl->set_mode(window, WINDOW_MODE_FULLSCREEN);
-	window->mode = WINDOW_MODE_FULLSCREEN;
+	else {
+		swc_window_set_geometry(base, &window->prev.geom);
+		window->mode = window->prev.mode;
+	}
 }
 
 EXPORT void
diff --git a/libswc/window.h b/libswc/window.h
index 08c9d36..d8f6bfa 100644
--- a/libswc/window.h
+++ b/libswc/window.h
@@ -52,6 +52,11 @@ struct window {
 	struct view_handler view_handler;
 	bool managed;
 	unsigned mode;
+	
+	struct {
+		struct swc_rectangle geom;
+		unsigned mode;
+	} prev;
 
 	struct {
 		struct window_pointer_interaction interaction;
-- 
2.49.0

