From 9f7ca588361f40444c10a039d32f541e378e2321 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Wed, 5 Jun 2024 01:59:02 +0200
Subject: [PATCH] remove arch4random_buf call. it does not exist

---
 Utilities/cmlibarchive/libarchive/archive_random.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Utilities/cmlibarchive/libarchive/archive_random.c b/Utilities/cmlibarchive/libarchive/archive_random.c
index a410dc08..6ba130ca 100644
--- a/Utilities/cmlibarchive/libarchive/archive_random.c
+++ b/Utilities/cmlibarchive/libarchive/archive_random.c
@@ -121,9 +121,6 @@ archive_random(void *buf, size_t nbytes)
 #elif !defined(HAVE_ARC4RANDOM_BUF) && (!defined(_WIN32) || defined(__CYGWIN__))
 	la_arc4random_buf(buf, nbytes);
 	return ARCHIVE_OK;
-#else
-	arc4random_buf(buf, nbytes);
-	return ARCHIVE_OK;
 #endif
 }
 
-- 
2.45.1

