diff --git a/Makelists b/Makelists
index ac54f6f..af430d6 100644
--- a/Makelists
+++ b/Makelists
@@ -272,37 +272,6 @@ JBIG2DEC_SRC += thirdparty/jbig2dec/jbig2_segment.c
 JBIG2DEC_SRC += thirdparty/jbig2dec/jbig2_symbol_dict.c
 JBIG2DEC_SRC += thirdparty/jbig2dec/jbig2_text.c
 
-# --- OPENJPEG ---
-
-OPENJPEG_CFLAGS += -Ithirdparty/openjpeg/src/lib/openjp2
-OPENJPEG_CFLAGS += -DOPJ_STATIC
-OPENJPEG_CFLAGS += -DOPJ_HAVE_INTTYPES_H
-OPENJPEG_CFLAGS += -DOPJ_HAVE_STDINT_H
-
-OPENJPEG_BUILD_CFLAGS += -Ithirdparty/openjpeg/src/lib/openjp2
-OPENJPEG_BUILD_CFLAGS += -DMUTEX_pthread=0
-
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/bio.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/cio.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/dwt.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/event.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/function_list.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/ht_dec.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/image.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/invert.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/j2k.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/jp2.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/mct.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/mqc.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/openjpeg.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/pi.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/sparse_array.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/t1.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/t2.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/tcd.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/tgt.c
-OPENJPEG_SRC += thirdparty/openjpeg/src/lib/openjp2/thread.c
-
 # --- FREEGLUT ---
 
 GLUT_CFLAGS += -Ithirdparty/freeglut/include
diff --git a/source/fitz/encode-jpx.c b/source/fitz/encode-jpx.c
index 40cca5a..e38604f 100644
--- a/source/fitz/encode-jpx.c
+++ b/source/fitz/encode-jpx.c
@@ -22,10 +22,10 @@
 
 #include "mupdf/fitz.h"
 
-#include <openjpeg.h>
-
 #if FZ_ENABLE_JPX
 
+#include <openjpeg.h>
+
 static opj_image_t *
 image_from_pixmap(fz_context *ctx, fz_pixmap *pix)
 {
