From 81bee8a1eb4bbaf1f515f386156b1a8a318dba16 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Thu, 15 Aug 2024 00:13:03 +0200
Subject: [PATCH 1/5] set tamzen as default font

---
 config.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 2cd740a..89fb5d1 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "Tamzen:pixelsize=14:antialias=true:autohint=true";
 static int borderpx = 2;
 
 /*
-- 
2.45.1

