From 666e445330cd1183beca770a691f94bed0651e8e Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Mon, 28 Aug 2023 21:44:42 +0000
Subject: [PATCH 2/5] set borderpx to 5

---
 config.def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 89fb5d1..d676f03 100644
--- a/config.def.h
+++ b/config.def.h
@@ -6,7 +6,7 @@
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
 static char *font = "Tamzen:pixelsize=14:antialias=true:autohint=true";
-static int borderpx = 2;
+static int borderpx = 5;
 
 /*
  * What program is execed by st depends of these precedence rules:
-- 
2.45.1

